### PR TITLE
Update dependencies: docker -> 7.0.0 and tabulate -> 0.9.0

### PR DESCRIPTION
**Pull request by danielhollas**
Main goal here is to upgrade to `docker-py==7.0.0` which contains a bugfix that I ran into while testing locally with Python 3.12:
https://github.com/docker/docker-py/issues/3151

Looks like the bug was introduced in version `6.2` which we bumped to in #194 but did not release yet so no users should be affected. 

Depends on #201

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -22,12 +22,11 @@ packages = find:
 install_requires =
     click==8.1
     click-spinner==0.1.10
-    docker==6.1.2
+    docker==7.0.0
     packaging==21.3
-    python-dotenv==0.19.1
```
Comment (danielhollas) [inline setup.cfg:27]: This doesn't seem to be used anywhere, we're using our own custom solution to store configuration for profiles

File: setup.cfg [hunk 2]
```diff
@@ -39,7 +38,6 @@ console_scripts =
 dev =
     bumpver==2023.1129
     dunamai==1.19.0
-pre_commit =
```
Comment (danielhollas) [inline setup.cfg:42]: Seems easier to fold this into dev. I am still keeping tests separate for CI convenience.

File: setup.cfg (hunk 1 above)
Comment (unkcpz) [reply setup.cfg:27]: Yep, agree.

File: .github/workflows/ci.yml
Comment (unkcpz) [inline .github/workflows/ci.yml:1]: The `pre-commit` job can be removed, we have the pre-commit.ci hook, this is duplicate check.

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks @danielhollas!
Just a small request. Can you delete the `pre-commit` from `ci.yml`?

> Looks like the bug was introduced in version 6.2

You mean `6.1.2` I guess?

File: .github/workflows/ci.yml
Comment (danielhollas) [reply .github/workflows/ci.yml:1]: Done

**Comment by danielhollas** [comment]
Thank you for the review. I removed the pre-commit action.

> You mean 6.1.2 I guess?
Ah, yep, my bad.

**Review by unkcpz — APPROVED** [review]
Looks all good! Thanks!

**Pull request merged**